src/compiler.jl

```diff
@@ -2680,11 +2770,16 @@ function julia_deallocator(B, Obj)
     Obj = LLVM.Value(Obj)
     mod = LLVM.parent(LLVM.parent(position(B)))
     ctx = context(mod)
-    ptr8 = LLVM.PointerType(LLVM.IntType(8; ctx))
+    
     T_void = LLVM.VoidType(ctx)
-    freeF = get_function!(mod, "free", LLVM.FunctionType(T_void, [ptr8]))
-    callf = call!(B, freeF, [pointercast!(B, Obj, ptr8)])
-    LLVM.API.LLVMAddCallSiteAttribute(callf, LLVM.API.LLVMAttributeIndex(1), EnumAttribute("nonnull"; ctx))
+    if any_jltypes(LLVM.llvmtype(Obj))
+        callf = LLVM.UndefValue(T_Void)
```

Comment:
you should just return LLVM.Value(NULL) here

Reply:
I think I tried this previously and that didn't work

Reply:
For freeing that should work, if not thats a bug on my end. For malloc, returning null is an error.

Reply:
There is a Isa check on the free in Enzyme proper

Reply:
Restore the return void and I'll change that do a != null and isa